sync-core/src/main/java/com/cloudant/sync/datastore/DocumentNotFoundException.java

```diff
@@ -19,13 +19,21 @@
 /**
  * Thrown when a document cannot be found.
  */
-public class DocumentNotFoundException extends RuntimeException {
+public class DocumentNotFoundException extends DocumentException {
+
+    public DocumentNotFoundException() {
+
+    }
```

Comment:
Seems like we could have a `docId, revId` and `docId` constructors which automatically create appropriate messages and call super appropriately rather than creating the message every time.